src/MassTransit/Middleware/ReceiveEndpointDependencyFilter.cs

```diff
@@ -0,0 +1,32 @@
+namespace MassTransit.Middleware
+{
+    using System.Threading.Tasks;
+    using Internals;
+    using Transports;
+
+
+    public class ReceiveEndpointDependencyFilter<TContext> :
+        IFilter<TContext>
+        where TContext : class, PipeContext
+    {
+        readonly ReceiveEndpointContext _context;
+
+        public ReceiveEndpointDependencyFilter(ReceiveEndpointContext context)
+        {
+            _context = context;
+        }
+
+        public async Task Send(TContext context, IPipe<TContext> next)
+        {
+            await _context.DependenciesReady.OrCanceled(context.CancellationToken).ConfigureAwait(false);
+
+            await next.Send(context).ConfigureAwait(false);
+        }
+
+        public void Probe(ProbeContext context)
+        {
+            var scope = context.CreateScope("receive-endpoint-dependencies");
```

Comment:
This won't format nicely in JSON, can you CamelCase it?

Reply:
InMemory and GRPC are exceptional, they have only transports:
https://github.com/MassTransit/MassTransit/blob/a3d66e5af78931e5baee1edcc8ca58221ae85731/src/MassTransit/InMemoryTransport/InMemoryTransport/InMemoryReceiveTransport.cs#L142